Mathlib/AlgebraicTopology/SimplicialSet/HomotopyCat.lean

```diff
@@ -0,0 +1,469 @@
+is not used for the construction of the homotopy category) and then composing with an analogously
+defined `SSet.hoFunctor₂ : SSet.Truncated.{u} 2 ⥤ Cat.{u,u}` implemented relative to the syntax of
+the 2-truncated simplex category.
+
+It should be relatively straightforward to show that these constructions agree:
+
+def hoFunctor.ofTwoTruncation.iso (V : SSet) :
+    SSet.hoFunctor₂Obj ((truncation 2).obj V) ≅ SSet.hoCat V := sorry
+
+def hoFunctor.ofTwoTruncation.natIso :
+     truncation 2 ⋙ SSet.hoFunctor₂ ≅ SSet.hoFunctor' := sorry
+
+but we leave this for future work.
+
+The functor `SSet.hoFunctor` is shown to be left adjoint to the nerve by providing an analogous
+decomposition of the nerve functor and then composing a pair of adjunctions, which factor through
+the category of 2-truncated simplicial sets.
+-/
+
+namespace CategoryTheory
+open Category Limits Functor Opposite Simplicial
+universe v u
+
+section
+
+local notation3:1000 (priority := high) X " _[" n "]" =>
+    (X : CategoryTheory.SimplicialObject _).obj (Opposite.op (SimplexCategory.mk n))
+
+/-- A simplicial set `S` has an underlying refl quiver with `S _[0]` as its underlying type.-/
+def SSet.OneTruncation (S : SSet) := S _[0]
+
+/-- The source vertex of `f : S _[1]` for use in defining the underlying refl quiver.-/
+def SSet.OneTruncation.src {S : SSet} (f : S _[1]) : OneTruncation S :=
+  S.map (SimplexCategory.δ (n := 0) 1).op f
+
+/-- The target vertex of `f : S _[1]` for use in defining the underlying refl quiver.-/
+def SSet.OneTruncation.tgt {S : SSet} (f : S _[1]) : OneTruncation S :=
+  S.map (SimplexCategory.δ (n := 0) 0).op f
+
+/-- The hom-types of the refl quiver underlying a simplicial set `S` are subtypes of `S _[1]`.-/
+def SSet.OneTruncation.Hom {S : SSet} (X Y : OneTruncation S) :=
+  {p : S _[1] // src p = X ∧ tgt p = Y}
+
+/-- A simplicial set `S` has an underlying refl quiver `SSet.OneTruncation S`.-/
+instance (S : SSet) : ReflQuiver (SSet.OneTruncation S) where
+  Hom X Y := SSet.OneTruncation.Hom X Y
+  id X := by
+    refine ⟨S.map (SimplexCategory.σ (n := 0) 0).op X, ?_, ?_⟩ <;>
+    · change (S.map _ ≫ S.map _) X = X
+      rw [← map_comp, (_ : _ ≫ _ = 𝟙 _)]; simp
+      show ({..} : Opposite _) = _; congr; ext i
+      let 0 := i
+      rfl
+
+/-- The functor that carries a simplicial set to its underlying refl quiver.-/
+def SSet.oneTruncation : SSet.{u} ⥤ ReflQuiv.{u,u} where
+  obj S := ReflQuiv.of (OneTruncation S)
+  map {S T} F := {
+    obj := F.app (op [0])
+    map := fun f => by
+      refine ⟨F.app (op [1]) f.1, ?_, ?_⟩
+      · change (F.app _ ≫ _) _ = _
+        rw [← F.naturality]
+        exact congrArg (F.app _) f.2.1
+      · change (F.app _ ≫ _) _ = _
+        rw [← F.naturality]
+        exact congrArg (F.app _) f.2.2
+    map_id := fun X => by
+      change ({..} : Subtype _) = {..}
+      congr
+      change _ = (F.app _ ≫ _) _
+      rw [← F.naturality]
+      rfl
+  }
+  map_id X := by rfl
+  map_comp f g := by rfl
```

Comment:
Automation should take care of this:
```suggestion
```
If any case, adding a lemma like this would facilitate automation:
```lean
@[ext]
lemma hom_ext {S : SSet} {x y : SSet.OneTruncation S} {f g : x ⟶ y}
    (h : f.1 = g.1) : f = g := by
  cases f; cases g; subst h; rfl
```